yapapi/rest/activity.py

```diff
@@ -187,11 +203,16 @@ async def __aiter__(self) -> AsyncIterator[events.CommandEventContext]:
             for result in results:
                 any_new = True
                 assert last_idx == result.index, f"Expected {last_idx}, got {result.index}"
-                if result.result == "Error":
-                    raise CommandExecutionError(result.message, last_idx)
 
-                message = dict(stdout=result.stdout, stderr=result.stderr, message=result.message)
-                kwargs = dict(cmd_idx=result.index, message=json.dumps(message, indent=4))
+                message = None
+                if result.message:
+                    message = result.message
+                elif result.stdout or result.stderr:
+                    message = json.dumps({"stdout": result.stdout, "stderr": result.stderr})
```

Comment:
Here we make sure that `message` is just `None` and not something like `{"stdout": None, "stderr": None}` so we can just ignore it e.g. the summary logger.